### PR TITLE
RSDK-7011, RSDK-7030 - Update initial config flow

### PR DESCRIPTION
**Pull request by cheukt**
update so that we only shorten timeout to 1s if cached config exists. Prevents slow I/O from stopping server startup
also log the last modified time for cached config if we use it to start up
```
2024-03-25T18:50:39.112Z        WARN    robot_server    config/reader.go:627    unable to get cloud config; using cached version        {"config last updated":"2024-03-25T14:50:23-04:00","error":"context deadline exceeded"}
```
no new tests because there isn't a way to ensure that timeout changed (hard to test that timeout is between 1 and 5s, for example) without running into possible race conditions

tested locally, seems correct

### CODE REVIEW

File: config/reader.go [hunk 1]
```diff
@@ -621,7 +619,12 @@ func getFromCloudOrCache(ctx context.Context, cloudCfg *Cloud, shouldReadFromCac
 				// return cache err
 				return nil, cached, cacheErr
 			}
-			logger.Warnw("unable to get cloud config; using cached version", "error", err)
+
+			lastUpdated := "unknown"
+			if fInfo, err := os.Stat(getCloudCacheFilePath(cloudCfg.ID)); err == nil {
+				lastUpdated = fInfo.ModTime().Format(time.RFC3339)
+			}
+			logger.Warnw("unable to get cloud config; using cached version", "config last updated", lastUpdated, "error", err)
```
Comment (benjirewis) [inline config/reader.go:627]: Nice, last updated is helpful. What does this log look like? /Why are we using that RFC3339 format?

**Review by benjirewis — COMMENTED** [review]
Generally LGTM, just a question about that last updated value.

File: config/reader.go (hunk 1 above)
Comment (cheukt) [reply config/reader.go:627]: sample log in description, time is formatted like`"config last updated":"2024-03-25T14:50:23-04:00"`. RFC3339 just happens to be what we seem to be using in other areas of the codebase (mostly in the CLI). happy to change if you have suggestions

Comment (cheukt) [reply config/reader.go:628]: sure, updated

**Review by benjirewis — APPROVED** [review]
LGTM! Thank you :)

**Pull request merged**